src/main.rs

```diff
@@ -61,14 +66,20 @@ fn main() {
           panic!("Could not find output path {}", output.display());
      }
 
-     let _threads = value_t!(matches, "threads", u32).unwrap_or_else(|e| e.exit());
-     let tile_chunk = value_t!(matches, "tile-chunk", usize).unwrap_or_else(|e| e.exit());
-     let _split_lanes = matches.is_present("split-lanes");
 
-     let novaseq_run = match novaseq_run::NovaSeqRun::read_path(run_path, tile_chunk) {
+    let _threads = value_t!(matches, "threads", u32).unwrap_or_else(|e| e.exit());
+    let tile_chunk = value_t!(matches, "tile-chunk", usize).unwrap_or_else(|e| e.exit());
+    let _split_lanes = matches.is_present("split-lanes");
+
+    let novaseq_run = match novaseq_run::NovaSeqRun::read_path(run_path, tile_chunk) {
           Ok(n_run) => n_run,
           Err(e) => panic!("Error reading NovaSeq run: {}", e),
      };
 
-     extract_reads::extract_samples(novaseq_run, output).unwrap();
+    extract_reads::extract_samples(novaseq_run, output).unwrap();
+        
+    let sample_data = match sample_data::SampleData::read_path(samplesheet.to_path_buf()) {
```

Comment:
```suggestion
    let _sample_data = match sample_data::SampleData::read_path(samplesheet) {
```
`samplesheet` is already a `PathBuf`. Prefixing the result with an underscore to avoid a warning.